dask/array/reductions.py

```diff
@@ -722,3 +723,70 @@ def validate_axis(ndim, axis):
         return axis + ndim
     else:
         return axis
+
+
+def topk(a, k, axis=-1, split_every=None):
+    """Extract the k largest elements from a on the given axis,
+    and return them sorted from largest to smallest.
+    If k is negative, extract the -k smallest elements instead,
+    and return them sorted from smallest to largest.
+
+    This assumes that ``k`` is small.  All results will be returned in a single
+    chunk along the given axis.
+
+    Examples
+    --------
+    >>> import dask.array as da
+    >>> x = np.array([5, 1, 3, 6])
+    >>> d = da.from_array(x, chunks=2)
+    >>> d.topk(2).compute()
+    array([6, 5])
+    >>> d.topk(-2).compute()
+    array([1, 3])
+    """
+    if isinstance(a, int) and isinstance(k, Array):
+        warnings.warn("DeprecationWarning: topk(k, x) has been replaced with topk(a, k)")
```

Comment:
Should be `topk(k, a) has bee been replaced with topk(a, k)`

Reply:
No, the previous API really was topk(k, x). I renamed the array to `a` to be coherent with all other reduction functions.

Reply:
Yes, but since the argument was positional, as a reader all I really care about is the swapping of the parameter order (array and `k`). As it currently reads it's not immediately clear that the order of the array and `k` parameter have swapped.